jdaviz/core/template_mixin.py

```diff
@@ -1901,6 +1909,162 @@ def __init__(self, *args, **kwargs):
+        # add marks to any new viewers
+        self.hub.subscribe(self, ViewerAddedMessage, handler=self._update_mark_coords)
+        # update coordinates when link type or reference data is _mode_changed
+        self.hub.subscribe(self, LinkUpdatedMessage, handler=self._update_mark_coords)
+
+    def _plugin_active_changed(self, *args):
+        for mark in self.marks:
+            mark.visible = self.plugin.is_active
+
+    @property
+    def image_viewers(self):
+        return [viewer for viewer in self.app._viewer_store.values()
+                if isinstance(viewer, BqplotImageView)]
+
+    @property
+    def marks(self):
+        # NOTE: this will require additional logic if we want to support multiple independent
+        # ApertureSubsetSelect instances per-plugin
+        all_aperture_marks = []
+        for viewer in self.image_viewers:
+            # search for existing mark
+            matches = [mark for mark in viewer.figure.marks
+                       if isinstance(mark, ApertureMark)]
+            if len(matches):
+                all_aperture_marks += matches
+                continue
+
+            x_coords, y_coords = self._get_mark_coords(viewer)
+
+            mark = ApertureMark(
+                viewer,
+                x=x_coords,
+                y=y_coords,
+                colors=['#c75109'],
+                fill_opacities=[0.0],
+                visible=self.plugin.is_active)
+            all_aperture_marks.append(mark)
+            viewer.figure.marks = viewer.figure.marks + [mark]
+        return all_aperture_marks
+
+    def _get_mark_coords(self, viewer):
+        if not len(self.selected) or not len(self.dataset.selected):
+            return [], []
+
+        if self.multiselect:
+            # assume first dataset (for retrieving the region object)
+            # but iterate over all subsets
+            spatial_regions = [self._get_spatial_region(dataset=self.dataset.selected[0], subset=subset)  # noqa
+                               for subset in self.selected]
+        else:
+            # use cached version
+            spatial_regions = [self.selected_spatial_region]
+
+        x_coords, y_coords = np.array([]), np.array([])
+        for spatial_region in spatial_regions:
+            if isinstance(spatial_region, PixelRegion):
+                # make a copy so changing the radius doesn't change the cached version in memory
+                pixel_region = spatial_region.copy()
+            else:
+                wcs = getattr(viewer.state.reference_data, 'coords', None)
+                if wcs is None:
+                    return [], []
+                pixel_region = spatial_region.to_pixel(wcs)
+            # NOTE: this assumes that we'll apply the same radius factor to all subsets (all will
+            # be defined at the same slice for cones in cubes)
+            pixel_region.radius *= self.radius_factor
+            roi = regions2roi(pixel_region)
+            x, y = roi.to_polygon()
```

Comment:
Why can't we grab the app `subset_state.roi` directly and call `to_polygon` on that? Then we don't have to do `roi -> region -> possibly another region -> roi -> to_polygon`.

Reply:
Theoretically glue would have already done all the translations for us across layers and each layer should have its own subset_state or something, right? (That's why we see same subset offset a little w.r.t. itself in some cases, no?)

Reply:
I think that did manage to simplify the logic - thanks!  Let me know if you see anything else!

Reply:
I think this was needed to handle change in WCS correctly.  I'll revert to the old version and if you have any ideas to simplify but still work on a change to rotation, feel free to try and push directly!